src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/GatherVcfs.java

```diff
@@ -91,8 +91,10 @@ protected Object doWork() {
     /** Checks (via filename checking) that all files appear to be block compressed files. */
     private boolean areAllBlockCompressed(final List<File> input) {
         for (final File f : input) {
-            // TODO move/expose these methods elsewhere, since VariantContextWriterFactory is deprecated
-            if (VariantContextWriterFactory.isBCFOutput(f) || !VariantContextWriterFactory.isCompressedVcf(f)) return false;
+            if (f != null) {
```

Comment:
You can use a `@DataProvider` for the test to make it clean and compact.  Come find me if you need help.